gemfiles/minimum_rubocop.gemfile

```diff
@@ -4,7 +4,11 @@ eval_gemfile "../Gemfile"
 
 # This naively extracts the minimum rubocop version from our gemspec,
 # allowing us to ensure that it remains compatible.
-minimum_rubocop_version = File.read("rubocop-shopify.gemspec")[/(?<=add_dependency\("rubocop", "~> ).*(?="\)$)/]
-raise "Failed to extract minimum rubocop version from gemspec" if minimum_rubocop_version.nil?
+version_match = File.read("rubocop-shopify.gemspec").match(/add_dependency\("rubocop",[^)]+\)/)[0]
+raise "Failed to extract rubocop version requirements from gemspec" if version_match.nil?
 
-gem "rubocop", minimum_rubocop_version
+# Parse version requirements and get the highest minimum version
+requirements = version_match.scan(/\d+\.\d+/)
+minimum_version = requirements.max
+
+gem "rubocop", "~> #{minimum_version}.0"
```

Comment:
We don't want to use `~>`, because that doesn't guarantee that we're using the exact minimum version. For example, this results in `~> 1.72.0` which would likely result in us installing `1.72.2`, whereas the actual minimum version we should be testing against is `1.71.1`.